src/mlpack/core/tree/vantage_point_tree/dual_tree_traverser_impl.hpp

```diff
@@ -0,0 +1,567 @@
+/**
+ * @file dual_tree_traverser_impl.hpp
```

Comment:
Is there any difference between this traverser and the BinarySpaceTree traverser?  If not, maybe we could make that traverser generic (accepting any tree type) and use it inside each class.  The same applies for the single-tree traverser.